### ISSUE TITLE
Array properties on _valuesEnd missing from _valuesStart adds `undefined` to property

### ISSUE THREAD

**Issue by msbit** [tweenjs/tween.js#446]
If the argument to `end` contains an array property that does not exist on the argument to `constructor`, the array is modified, prepending`undefined`. This bit me as I was passing an object through and was not expecting it to be modified, and in particular didn't expect for there to be an `undefined` in the array :) 

This happens at:
https://github.com/tweenjs/tween.js/blob/706786f5561b05fe992f56dcb9115f1ebd5cdf86/src/Tween.js#L190

### Tween.js version
* master (706786f5561b05fe992f56dcb9115f1ebd5cdf86)

### Javascript Environments:
* Safari Version 12.1.1 (14607.2.6.1.1)
* Chrome Version 74.0.3729.169 (Official Build) (64-bit)
* NodeJS v10.15.3

### Reproduction:

```
const TWEEN = require('./tween.js/src/Tween.js');

const start = { x: 10, y: 20 };

const end = { x: 100, y: 200, values: ['a', 'b'] };

new TWEEN.Tween(start).to(end).start();

console.log(end);
```

### Expected Output:

```
{ x: 100, y: 200, values: [ 'a', 'b' ] }
```

### Actual Output:

```
{ x: 100, y: 200, values: [ undefined, 'a', 'b' ] }
```

**Comment by mikebolt** [comment]
That is bad. I will work on this. I need to refactor some code.

**Comment by mikebolt** [comment]
Okay, this was an easy fix. Thanks for the report. Go checkout the repo here and test it: https://github.com/mikebolt/tween.js/tree/mikebolt-end-unmodified

The PR is #447, I will close this once it is merged into master and updated in NPM.

**Comment by msbit** [comment]
That branch looks good from my testing.

In my project I'd already moved to passing a copy of the long-lived object to `to()`, so I'm not able to easily test that out, but I also tested your fix using an instance of a `class`:

```
class End {
  constructor(x, y, values) {
    this.x = x;
    this.y = y;
    this.values = values;
  }   
}

const start = { x: 10, y: 20 };
const end = new End(100, 200, ['a', 'b']);
new TWEEN.Tween(start).to(end).start();
console.log(end);
```

which also gives me the correct output:

```
End { x: 100, y: 200, values: [ 'a', 'b' ] }
```

**Comment by mikebolt** [comment]
merged
